Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -97,25 +97,25 @@ fileprivate extension Compiler.ByteCodeGen {
     switch kind {
     case .startOfSubject:
       builder.buildAssert { (input, pos, bounds) in
-        pos == input.startIndex
+        pos == bounds.lowerBound
```

Comment:
Is the bounds here the search bounds or subject bounds?

Reply:
Also, FWIW, I'm going to try to have @rctcwyvrn make a builtin assertion and/or character class instruction, so all of these closures will just be switch case bodies in the matching engine fairly soon.

Reply:
Let's do the built in assertion change first (as in soon)